attachment_s3/models/ir_attachment.py

```diff
@@ -191,3 +199,11 @@ def _store_file_delete(self, fname):
                     )
         else:
             super()._store_file_delete(fname)
+
+    def unlink(self):
+        for attachment in self:
+            if not self.env.context.get('force_s3_unlink') and attachment.store_fname.startswith('s3://') and int(self.env['ir.config_parameter'].sudo().get_param('attachment_s3.retention_days', '0')):
```

Comment:
`            if not attachment.to_delete and attachment.store_fname.startswith('s3://') and int(self.env['ir.config_parameter'].sudo().get_param('attachment_s3.retention_days', '0')):
`

y en el cron no hace falta incorporar la clave 'force_s3_unlink', simplemente buscamos los to_delete